src/raft_log.rs

```diff
@@ -272,14 +282,15 @@ impl<T: Storage> RaftLog<T> {
         self.applied
     }
 
-    /// Attempts to set the stable up to a given index.
-    pub fn stable_to(&mut self, idx: u64, term: u64) {
-        self.unstable.stable_to(idx, term)
+    /// Returns the unstable entries and moves the stable offset up to the
+    /// last index, if there is any.
+    pub fn stable_entries(&mut self) -> Vec<Entry> {
```

Comment:
I think we should reuse the previous `stable_to` and `stable_snap_to` functions, these two functions called after entries and/or snapshot did stable, and which means we can get them from the `Storage` interface, so we can remove them from `unstable`. Otherwise, if we call `stable_entries` and/or `stable_snap` and before we actually stable them to `Storage`, any calls (outside of `raft-rs` by the applications) of functions like `last_index`,  `last_term`, etc. will return a false result.